spec/operators/catch-spec.ts

```diff
@@ -242,4 +242,15 @@ describe('Observable.prototype.catch', () => {
       done();
     });
   });
+
+  it('should allow types with operators such as mergeMap to flow correctly', () => {
+    type(() => {
+      Observable.from([{ a: 1 }, { a: 2 }, { a: 3} ])
+          .mergeMap(value => {
+              return Observable.of([value]);
+          })
+          .mergeMap(x => x)
+          .subscribe(x => x.a);
+    });
+  });
```

Comment:
In that case isn't this something in compiler behavior (regression maybe?) similar to https://github.com/ReactiveX/rxjs/issues/1806 (though it was babel side) ?